### ISSUE TITLE
[rspamd] fuzzy hashes

### ISSUE THREAD

**Issue by hoellen** [Mailu/Mailu#456]
Hello,

I am getting spam on my mailu instance. If I move it to the spam folder rspamd doesn't learn from it, because there are too few words (tokens) in this mail. You can see it in the logs:

```
antispam_1   | 2018-04-18 05:50:44 #7(controller) <130b34>; csession; rspamd_stat_classifiers_learn: <eztawyj57055343.62511464@mail.installio.co.ua> contains less tokens than required for bayes classifier: 4 < 11
antispam_1   | 2018-04-18 05:50:44 #7(controller) <130b34>; csession; rspamd_task_process: learn error: <eztawyj57055343.62511464@mail.installio.co.ua> contains less tokens than required for bayes classifier: 4 < 11
antispam_1   | 2018-04-18 05:50:44 #7(controller) <130b34>; csession; rspamd_controller_learn_fin_task: cannot learn <eztawyj57055343.62511464@mail.installio.co.ua>: <eztawyj57055343.62511464@mail.installio.co.ua> contains less tokens than required for bayes classifier: 4 < 11
```

To get rid of this annoying spam, we can use `fuzzy hashes` in rspamd. 

> Fuzzy hashes are used to search for similar messages – i.e. you can find messages with the same or a slightly modified text using this method. This technology fits well for blocking spam that is simultaneously sent to many users. Since the hash function is unidirectional, it is impossible to restore the original text using a hash only. And this allows you to send requests to third-party hash storages without risk of disclosure.

Information:
[\[1\] Fuzzy check](https://rspamd.com/doc/modules/fuzzy_check.html)
[\[2\] Usage of fuzzy hashes](https://rspamd.com/doc/fuzzy_storage.html)

It seems that this feature is disabled by default. So it would be nice to use this in mailu, so we have better spam filtering. 

hoellen

**Comment by kaiyou** [comment]
Agreed, I will try and enable them on master.

**Comment by kaiyou** [comment]
It would appear this is part the cause of the mayhem regarding worker-fuzzy. Investigation is in progress.

**Comment by Harti** [comment]
Has there been any progress on this investigation?
For some reason, the spam daemon seems to want to have 1,000 spam mails before it tries to detect any pattern. At this rate, I will receive and move spam mails to Junk for the next... 200 days. This is unbearable.

**Comment by ofthesun9** [comment]
Fixed by #527
